services/runservice/types/executortask.go

```diff
@@ -53,6 +54,8 @@ type ExecutorTaskSpec struct {
 
 	// Stop is used to signal from the scheduler when the task must be stopped
 	Stop bool `json:"stop,omitempty"`
+	// Timeout is used to signal from the scheduler when the task is on timeout
+	Timeout bool `json:"timeout,omitempty"`
```

Comment:
We could just use Stop to signal the task shoud be stopped and add a StopReason enum (values: `RunStopped`, `TaskTimedout` so the Executor could know why it should stop it and correctly set the task status.